### PR TITLE
Add VCAP_SERVICES env vars to LRPs

### LINKED ISSUE #462: [Feature]: App Developer can see Service Credential Bindings projected onto running workloads via `VCAP_SERVICES`
### Blockers/Dependencies

_No response_

### Background

**As an** App Developer
**I want** Credentials from Service Bindings for User-provided Service Instances to be accessible via the `VCAP_SERVICES` environment variable
**So that** my existing CF apps that are `VCAP_SERVICES` aware can continue to work

---

### Acceptance Criteria

**GIVEN** I have multiple `CFServiceBindings` bindings bound to an app
**WHEN I** inspect the running `Pods` for that app
**THEN I** see that there is a `VCAP_SERVICES` environment variable with its value set to a json string containing credentials/info about each binding.

For example, in CF today you'd see something like this after `cf ssh`-ing on to an app:

```
vcap@ea6b07a8-bb91-4b16-43d3-9b4e:~$ env | grep VCAP_SERVICES -A 18
VCAP_SERVICES={"user-provided":[{
  "label": "user-provided",
  "name": "test-upsi",
  "tags": [

  ],
  "instance_guid": "f4a3a20f-8fd5-4233-afd3-cfcd863986e1",
  "instance_name": "test-upsi",
  "binding_guid": "13bbc5ee-14a5-400b-a354-a68dd52a1807",
  "binding_name": null,
  "credentials": {
    "user-provided-key-1": "user-provided-value-1",
    "user-provided-key-2": "user-provided-value-2"
  },
  "syslog_drain_url": null,
  "volume_mounts": [

  ]
}]}
```

---

### Dev Notes

* `"name"` is an optional field for bindings. When `"name"` is empty/missing it should be set to be the same name as the Service Instance in `VCAP_SERVICES`
* In the exploration/spike we found that the the `ProcessController` was the best place to do this

Resources:
* Related Exploration: https://github.com/cloudfoundry/cf-k8s-controllers/issues/367
* [UPSI Proposal Doc](https://docs.google.com/document/d/1pi7u2zH07oeKS6TalWhzsnFPwbCy44SrvPGRMfomops/edit#heading=h.9w70n3ykktw2)

### PR DESCRIPTION
**Pull request by gnovv**
## Is there a related GitHub Issue?

resolves #462

## What is this change about?
Implements `VCAP_SERVICES` env vars in process controller.

## Does this PR introduce a breaking change?
No

## Acceptance Steps
See issue.

## Tag your pair, your PM, and/or team
@matt-royal

### CODE REVIEW

File: controllers/config/samples/cfdomain.yaml [hunk 1]
```diff
@@ -4,4 +4,4 @@ metadata:
   name: 5b5032ab-7fc8-4da5-b853-821fd1879201
   namespace: cf
 spec:
-  name: vcap.me
+  name: apps.cranky-jellybean.k8s-dev.relint.rocks"
```
Comment (tcdowney) [inline controllers/config/samples/cfdomain.yaml:7]: Probably want to swap this back.

Comment (gnovv) [reply controllers/config/samples/cfdomain.yaml]: thank you!

Comment (matt-royal) [reply controllers/config/samples/cfdomain.yaml]: Thanks for catching this

**Review by matt-royal — APPROVED** [review]
We paired on this. Looks good to me.

**Pull request merged**